storage/engine.go

```diff
@@ -1,16 +1,61 @@
 package storage
 
-import "github.com/nuts-foundation/nuts-node/core"
+import (
+	"errors"
+	"github.com/nuts-foundation/nuts-node/core"
+	"github.com/nuts-foundation/nuts-node/storage/log"
+	"path"
+)
 
 // New creates a new instance of the storage engine.
-func New() core.Engine {
-	return &engine{}
+func New() Engine {
+	return &engine{stores: map[string]KVStore{}}
 }
 
 type engine struct {
+	datadir string
+	stores  map[string]KVStore
 }
 
-// Name returns the name of the storage engine.
+// Name returns the name of the engine.
 func (e engine) Name() string {
 	return "Storage"
 }
+
+func (e engine) Start() error {
+	return nil
+}
+
+func (e engine) Shutdown() error {
+	failures := false
+	for storeName, store := range e.stores {
+		err := store.Close()
+		if err != nil {
+			log.Logger().Errorf("Failed to close store '%s': %s", storeName, err)
+			failures = true
+		}
+	}
+	if failures {
+		return errors.New("one or more stores failed to close")
+	}
+	return nil
+}
+
+func (e *engine) Configure(config core.ServerConfig) error {
+	e.datadir = config.Datadir
+	return nil
+}
+
+func (e *engine) GetKVStore(engineKey string, storeName string) (KVStore, error) {
```

Comment:
My guess is that this is not pluggable yet, but must be in the future. A TODO comment would give a heads up

Reply:
It will become pluggable by providing multiple implementations and a specific store to be configured (in the storage engine) given `engineKey` and `storeName`. Or at least that's my vision for it.

I don't think we need a TODO here, it's obvious you'd be choosing the store implementation to return here, depending the engine's configuration.

Reply:
Then `Engine` needs to be clarified a bit more. Is an engine something like Redis or based on the Nuts-go engine calling it?

Reply:
Latter one, so network module would pass in "network" and "connections". I'll rename it to something less specific and add some more godoc.